taichi/llvm/llvm_context.cpp

```diff
@@ -136,9 +136,9 @@ void compile_runtime_bitcode(Arch arch) {
     if (ret) {
       TI_ERROR("LLVMRuntime compilation failed.");
     }
-    std::system(fmt::format("llvm-as {}runtime.ll -o {}{}", runtime_folder,
```

Comment:
Thanks for fixing this UB here!

Reply:
Wait, what UB.. All I did was assigning this to `cmd`, then calling `std::system(cmd)`... I did that because i had to print out `cmd` 😂

Reply:
Here, in my understanding, the result of `fmt::format` will be destroyed after you ask for `.c_str()`. Then the `const char *` is actually a wild pointer...

Reply:
Hmm, https://blog.galowicz.de/2016/03/23/const_reference_to_temporary_object/. I feel like `~string` is invoked after `c_str()`, but i'm not too sure as well... Anyway, it should be correct now :)

Reply:
> I feel like `~string` is invoked after `c_str()`

That's true. But by the time you pass the `c_str()` pointer to `std::system`, the data pointed by the pointer should have been destroyed. This is the real UB I guess.

Reply:
Actually, https://stackoverflow.com/questions/10006891/stdstringc-str-and-temporaries, so the old usage was also safe.. ;)

Reply:
Oh, I see - I learned something today!! Thanks for correcting me!

Reply:
To be fair I also just learnt about that... I guess it's always easy to learn new things about C++..

Reply:
Right - C++ is too hard, and that's why we use it to create a compiler so that other people can use Taichi instead of C++... :-)